xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp

```diff
@@ -823,6 +824,8 @@ void CVideoPlayerVideo::ProcessOverlays(const VideoPicture* pSource, double pts)
     for(it = overlays.begin(); it != overlays.end(); ++it)
     {
       double pts2 = (*it)->bForced ? pts : pts - m_iSubtitleDelay;
+      CLog::Log(LOGINFO, "CHIAMATA m_renderManager qui: m_iSubtitleDelay {}", m_iSubtitleDelay);
```

Comment:
Italian comment :)
Can you explain why `subtitleOffset` was added here? Also please name all member functions as `m_*`

Reply:
whoops :D oversight
several Kodi developers have told me in the past to remove prefixes in new variables
here is different thing? only to understand

Reply:
I guess you might be confusing it with the hungarian notation (e.g. `m_ievents` to specify an integer) and not exactly the member identifier. But I can be wrong as well :)

Reply:
oh O_O i don't remember
no problem

Reply:
> Can you explain why subtitleOffset was added here? 

I had only found this way to be able to modify the PTS of the libass renderer 
the current implementation IMO suspect has never worked with libass
before the overlay contained the data to be rendered (like text) now the rendering it is independent from the overlay
libass renderer is based on the current PTS and not on the start / stop values of the overlay
this should in theory only cause problems with libass rendering

however i have removed all "offset" change,
i do not know how to test if really works,
maybe you could take a look on this, to be fix in another PR,
since this my suspected problem was already present in the official kodi anyway

Reply:
This is exactly one of my concerns with your new implementation - I recognize some merits on the old overlay implementation. By removing the overlay and using ass events directly you'll loose the ability of having a container for your events that can carry not only the event but additional data. This can cause limitations in the future. Suppose that you want to have a vtt parser where each overlay can have a different position on the screen (just like Netflix does). Personally I feel the overlay would be the best place to carry and propagate that data in a format independent way. Forcing it in the ass event may be possible but I think it will always be too static. I mean, suppose that because of some sort of requirement, the positional data for the overlay must be recalculated because of a settings change. This is probably hard to do with events only as you'd have to iterate through all the events in the track and recalculate them

Reply:
IMO kodi overlay is only a surplus for libass case, libass already takes care of everything

> Suppose that you want to have a vtt parser where each overlay can have a different position on the screen (just like Netflix does)...

yes, but you are talking about converted subtitles formats that are special cases, with special settings

the position text values have to be fully managed by libass, attach these values to the overlays is not needed and IMO greatly increases the current code complexity and the amount of data to be processed at every frame
i also believe that the current problem of delay could increase

if the text change position for every Event you are assuming that we should create an Style for each Event in order to set the position, and i do not think this is a correct way to do this, because the styles should be used for more events not linked to one event, and this increase all complexity IMO unnecessarily.

theorizing of VTT case, we have text position values for each text entry, font color or other
we theorize to let to the user some special settings:
"Ignore VTT text positions" ON/OFF
"Ignore VTT font styles" ON/OFF

being a format type conversion, IMO the best way is set the text positions by using the forced ass tag positions
(is the reason because there is a fixed PlayRes values for "adapted" subtitles types:
https://github.com/CastagnaIT/xbmc/blob/subtitles_libass/xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.cpp#L174-L175)

then, if the user change a special VTT settings,
we should find a way to re-parse again the file by respecting the new settings,
or otherwise you can disable these special VTT settings when the player is in playing, which is not a great loss

Reply:
> if the text change position for every Event you are assuming that we should create an Style for each Event in order to set the position, and i do not think this is a correct way to do this,

if you are thinking to create a single Style and changes his values every frame, suggest to do not do this
libass developers warn to not alter styles, or do it as little as possible

Reply:
I am not saying the previous implementation is the way to go, I am just trying to ensure we can have the same functionality as before (now with libass) but without compromising the requirements of future features. Basically that we manage to have the foundations for extensibility right.

IMHO it's the overlayrenderer responsibility (which basically is a manager) to dictate how, where and with which style the libass handler should render an ass image.

 In the particular case of vtt I don't think creating a setting per format is the way to go as you suggest. We likely want a setting that applies to all sub formats that can render overlays at arbitrary positions (such as vtt or even ass). That  setting would behave similarly to the old "assfixedworks" advanced setting. If on will respect the fixed position defined in the calibration settings, otherwise it will respect the position dictated by the parser. 
Either way, we need to figure out a way of passing this data in a format and resolution independent way. Can be (and should be) done later in the scope of another PR, as long as we don't identify any limitation now.

Reply:
the screen resolution is not a problem, all text positions coordinates depends on the PlayResX/PlayResY values,
that has no relation with the actual screen resolution, even if it changes, because the scale of subtitles images are done after libass rendering

> We likely want a setting that applies to all sub formats that can render overlays at arbitrary positions (such as vtt or even ass)

yes agree my was only an simple example to try avoid misunderstoods

however, as the libass devs themselves have told me, the "overrides" for ass positions/styles like implemented in kodi are a kind of hack, is an extra feature that may not work in some particular cases,
in the same way override the original text positions of others formats (like VTT) means doing another type of hack

in any case, all the basic functionalities are all there, also if you cannot doing the overrides hacks (positions/styles) always in easy way